server/src/main/java/org/elasticsearch/index/seqno/ReplicationTracker.java

```diff
@@ -1348,6 +1348,10 @@ private synchronized void setHasAllPeerRecoveryRetentionLeases() {
         assert invariant();
     }
 
+    public synchronized boolean hasAllPeerRecoveryRetentionLeases() {
+        return hasAllPeerRecoveryRetentionLeases;
```

Comment:
Currently, `hasAllPeerRecoveryRetentionLeases` is initially defined as
```
this.hasAllPeerRecoveryRetentionLeases = indexSettings.getIndexVersionCreated().onOrAfter(Version.V_7_4_0)
```

I guess we need to add `&& softDeletesEnabled`?

Reply:
Good point. Fixed in https://github.com/elastic/elasticsearch/pull/49448/commits/748eeea7c735c1c2b5304439d5e28620132e3d42.